quesma/model/bucket_aggregations/dateRange.go

```diff
@@ -196,6 +196,9 @@ func (query DateRange) DoesNotHaveGroupBy() bool {
 func (query DateRange) CombinatorGroups() (result []CombinatorGroup) {
 	for intervalIdx, interval := range query.Intervals {
 		prefix := fmt.Sprintf("range_%d__", intervalIdx)
+		if len(query.Intervals) == 1 {
```

Comment:
So, it is a combinator approach that adds functions like `countIf` is incompatible with JOINs needed for `top_hits` and `top_metrics`.  I fixed it by flattening this typo of aggregations to one range and creating multiple SQLs.

However, than either needs to add special handling of column prefixes which is tricky in joins or simple skip prefixes if there is just one group. Since `filter` does not add prefix, I went with that approach.